applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -145,28 +144,57 @@ static void handle_dvfs_error(int32_t err)
 	(void) k_work_reschedule(&dvfs_retry.retry_work, timeout);
 }
 
-static void set_dvfs_freq(enum dvfs_frequency_setting target_freq)
+static void dvfs_notify_cb(void)
+{
+	int res;
+	int32_t ret = sys_notify_fetch_result(&cli.notify, &res);
+
+	if (ret < 0) {
+		LOG_DBG("work not completed");
```

Comment:
```suggestion
		LOG_DBG("Work not completed");
```

should such case happen? will this cb be executed if the operation is not yet ready?

Reply:
It should not happen.

This may only happen in case some other event on cli.notify happened.

There are also: SYS_NOTIFY_METHOD_SPINWAIT, SYS_NOTIFY_METHOD_SIGNAL, SYS_NOTIFY_METHOD_CALLBACK. They should not happen in our case

Reply:
then maybe there should be an assert/fatal error?

Reply:
If this happens it means some other module is also using cli.notify.

 I would drop such situation instead of failing whole system.